services/yaml/yaml.tester.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('../service-tester');
+const colorscheme = require('../../lib/colorscheme.json');
+const mapValues = require('lodash.mapvalues');
+
+const colorsB = mapValues(colorscheme, 'colorB');
+
+const t = new ServiceTester({ id: 'dynamic-yaml', title: 'User Defined YAML Source Data', pathPrefix: '/badge/dynamic/yaml' });
+module.exports = t;
+
+t.create('Connection error')
+  .get('.json?url=https://github.com/badges/shields/raw/master/package.json&query=$.name&label=Package Name&style=_shields_test')
+  .networkOff()
+  .expectJSON({ name: 'Package Name', value: 'inaccessible', colorB: colorsB.red });
+
+t.create('No URL specified')
+  .get('.json?query=$.name&label=Package Name&style=_shields_test')
+  .expectJSON({ name: 'Package Name', value: 'no url specified', colorB: colorsB.red });
+
+t.create('No query specified')
+  .get('.json?url=https://raw.githubusercontent.com/kubernetes/charts/568291d6e476c39ca8322c30c3f601d0383d4760/stable/coredns/Chart.yaml&label=Package Name&style=_shields_test')
```

Comment:
You've used a link to specified commit, which great in my opinion :-)

Reply:
👍 I thought it would be consistent when testing.